packages/database/src/migrations/20210524212900-FixEntityAggregationsLaosSchools-modifies-data.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+import { updateValues } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+async function getDashboardReportById(db, id) {
+  const { rows: dashboardReports } = await db.runSql(`
+      SELECT * FROM "dashboardReport"
+      WHERE id = '${id}';
+  `);
+  return dashboardReports[0] || null;
+}
+
+// Update data builder configuration for a report
+async function updateBuilderConfigByReportId(db, newConfig, reportId) {
+  return updateValues(db, 'dashboardReport', { dataBuilderConfig: newConfig }, { id: reportId });
+}
+
+const ENTITY_AGGREGATIONS_BY_REPORT_ID = {
+  Laos_Schools_Male_Female: {
+    dataSourceEntityType: 'school',
+    aggregationEntityType: 'country',
```

Comment:
So there's this crazy thing right, where actually you _can_ use `country` as your aggregation entity type, because when you're at the district level, the data source entities being passed in are only those for the district you're in, and they sum up to `LA` rather than `LA_XXX` but who cares, because the sum is the same! It's not ideal, and `requested` is absolutely the future way to go, but I thought this might be good enough for now.

However, you're actually right that I did make a big mistake and this doesn't work any more - at the country or any other level! `MOST_RECENT_PER_ORG_GROUP` is finding the most recent response across all schools, so we only get one school represented in the result, which is obviously not at all what we want. We can do `SUM_PER_ORG_GROUP` instead, and it actually works perfectly fine (with aggregation entity type set to `country`, still works fine at ever level), but it would break if other data were submitted in future, because those schools with multiple data points would have both counted.

What we really want is a `MOST_RECENT_PER_ENTITY` followed by a `SUM_PER_ORG_GROUP`. Unfortunately we're not going to get a heap of efficiency out of the db there, especially while there is only one submission per school, as the first aggregation won't cut down the results at all.

Still, we have to be correct, so I'm backing out this migration altogether 😞 

This makes me think maybe we do have to be less atomic with aggregation types, and allow e.g. a `SUM_PER_ORG_GROUP_MOST_RECENT_PER_ENTITY` 🙄